synapse/rest/client/v2_alpha/register.py

```diff
@@ -743,6 +729,62 @@ def _do_guest_registration(self, params, address=None):
         }))
 
 
+def _map_email_to_displayname(address):
+    """Custom mapping from an email address to a user displayname
+
+    Args:
+        address (str): The email address to process
+    Returns:
+        str: The new displayname
+    """
+    # Split the part before and after the @ in the email.
+    # Replace all . with spaces in the first part
+    parts = address.replace('.', ' ').split('@')
+
+    # Figure out which org this email address belongs to
+    org_parts = parts[1].split(' ')
+
+    # If this is a ...matrix.org email, mark them as an Admin
+    if org_parts[-2] == "matrix" and org_parts[-1] == "org":
+        org = "Tchap Admin"
+
+    # Is this is a ...gouv.fr address, set the org to whatever is before
+    # gouv.fr. If there isn't anything (a @gouv.fr email) simply mark their
+    # org as "gouv"
+    elif org_parts[-2] == "gouv" and org_parts[-1] == "fr":
+        org = org_parts[-3] if len(org_parts) > 2 else org_parts[-2]
+
+    # Otherwise, mark their org as the email's second-level domain name
+    else:
+        org = org_parts[-2]
+
+    def cap(s):
+        """Capitalise words in a string, including examples such as
+        'John-Doe'"""
+        if not s:
+            return s
+
+        # Convert str to a list so that we can edit each character
+        s = list(s)
```

Comment:
Alas no, you can't split strings like this because lol code points vs runes hahahah :sob: 

I think you probably want to split on whitespace, capitalise each, then join, and then do the same for hypens. BUT GOOD NEWS, that is equivalent to `string.capwords(string.capwords(s), "-")`

Reply:
Unfortunately it doesn't quite seem to be:

```
>>> a = 'jean-philippe martin-laval'
>>> string.capwords(string.capwords(a), "-")
'Jean-Philippe martin-Laval'
>>> 
```